### PR TITLE
[editorial] Add definitions for uniformity tags

### PR DESCRIPTION
**Pull request by alan-baker**
* A definitions for each tag type: call site, function and parameter
* Add tables for the possible values of each tag and the meaning
* Renames some of the tag values for clarity
* Update subsequent uses to link to the new defintions

### CODE REVIEW

File: wgsl/index.bs [hunk 1]
```diff
@@ -7103,32 +7103,73 @@ The second phase explores that graph, resulting in either rejecting the program,
...
+      <td>The parameter must be a [=uniform value=] in order for the [=return value=] to be a uniform value.
+  <tr><td><dfn noexport>ParameterNoRestriction</dfn>
+      <td>The parameter value has no uniformity requirement.
+</table>
+
+The following algorithm describes how to compute these tags for a given function:
 
-* Create nodes called "RequiredToBeUniform", "MayBeNonUniform", "CF_start", "CF_return", and if the function is non-void a node called "Value_return"
-* Create one node for each parameter of the function which we'll call "arg_i"
+* Create nodes called "RequiredToBeUniform", "MayBeNonUniform", "CF_start", "CF_return", and if the function is non-void a node called "Value_return".
```
Comment (dneto0) [inline wgsl/index.bs:7155]: instead of "the function is non-void".... say "if the funciton has a [=return type=]".

File: wgsl/index.bs [hunk 2]
```diff
@@ -7103,32 +7103,73 @@ The second phase explores that graph, resulting in either rejecting the program,
...
+  <tr><td><dfn noexport>CallSiteNoRestriction</dfn>
+      <td>The function may be called from [=uniform control flow|non-uniform control flow=].
+</table>
+
+<table class='data'>
+  <caption>[=Function tag=] values</caption>
+  <thead>
+    <tr><th>Function Tag<th>Description
+  </thead>
+  <tr><td><dfn noexport>SubsequentControlFlowMayBeNonUniform</dfn>
```
Comment (dneto0) [inline wgsl/index.bs:7130]: The "May" here is confusing about which entity is being constrained.
We're trying to say that executing the function may cause subsequent control flow to be nonuniform.
An alternative (to a naive reader), is that it is permissible to have non-uniform control flow executed after a call to this function.

One possibility is to rename the enums:

      SubsequentControlFlowMayBeNonUniform ->  MayCauseNonUniformControlFlow

      ReturnValueMayBeNonUniform -> MayReturnNonUniformValue

Either that, or change the description to say

     Calling this function may cause control flow to be nonuniform immediately after the call site.

I see there is some rhyming with the argument tags.  So maybe the best thing is to just update the description.

**Review by dneto0 — CHANGES_REQUESTED** [review]
This is a nice, mostly mechanical change.

I have two small requests

**Review by ben-clayton — COMMENTED** [review]
LGTM, thanks!

**Pull request merged**